cmd/clusterctl/api/v1alpha3/labels.go

```diff
@@ -22,4 +22,8 @@ const (
 
 	// ClusterctlCoreLabelName defines the label that is applied to all the core objects managed by clusterctl.
 	ClusterctlCoreLabelName = "clusterctl.cluster.x-k8s.io/core"
+
+	// ClusterctlSharedResourceLabelName defines the label that is applied to all the objects that are shared between
+	// instances of the same provider. e.g. CRDs, ValidatingWebhookConfiguration, MutatingWebhookConfiguration etc.
+	ClusterctlSharedResourceLabelName = "clusterctl.cluster.x-k8s.io/shared"
```

Comment:
```suggestion
	ClusterctlSharedResourceLabelName = "clusterctl.cluster.x-k8s.io/lifecycle"
```

clusterctl.cluster.x-k8s.io/lifecycle: "shared"

```
// ResourceLifecycle configures the lifecycle of a resource
type ResourceLifecycle string

const (
	// ResourceLifecycleOwned is the value we use when tagging resources to indicate
	// that the resource is considered owned and managed by the cluster,
	// and in particular that the lifecycle is tied to the lifecycle of the cluster.
	ResourceLifecycleOwned = ResourceLifecycle("owned")

	// ResourceLifecycleShared is the value we use when tagging resources to indicate
	// that the resource is shared between multiple clusters, and should not be destroyed
	// if the cluster is destroyed.
	ResourceLifecycleShared = ResourceLifecycle("shared")
```